datafusion/expr/src/expr_schema.rs

```diff
@@ -198,6 +202,11 @@ impl ExprSchemable for Expr {
                 let data_type = expr.get_type(input_schema)?;
                 get_indexed_field(&data_type, key).map(|x| x.is_nullable())
             }
+            Expr::GroupingSet(_) => {
+                // grouping sets do not really have the concept of nullable and do not appear
```

Comment:
I wonder if it is worth considering representing the grouping sets in a more specific way (like maybe a specific list on `LogicalPlan::Aggregrate`) rather than an `Expr` that can appear anywhere



```rust
enum GroupingExpr {
  Expr(Expr),
  GroupingSet(GroupingSet),
}

/// Aggregates its input based on a set of grouping and aggregate
/// expressions (e.g. SUM).
#[derive(Clone)]
pub struct Aggregate {
    /// The incoming logical plan
    pub input: Arc<LogicalPlan>,
    /// Grouping expressions
    pub group_expr: Vec<AggExprs>,
    /// Aggregate expressions
    pub aggr_expr: Vec<Expr>,
    /// The schema description of the aggregate output
    pub schema: DFSchemaRef,
}
```

Or something?